dask/dataframe/io/json.py

```diff
@@ -170,13 +170,15 @@ def read_json(url_path, orient='records', lines=None, storage_options=None,
 def read_json_chunk(chunk, encoding, errors, kwargs, meta=None):
     s = io.StringIO(chunk.decode(encoding, errors))
     s.seek(0)
-    df = pd.read_json(s, orient='records', lines=True, **kwargs)
+    read_function = kwargs.pop('read_function', pd.read_json)
```

Comment:
Can you add this name as a regular keyword argument, rather than popping from kwargs? The docs say we pass kwargs through to pandas.

This is somewhat similar to our `engine` argument in things like `read_parquet`. The difference is that this keyword changes the reader *and* the type of concrete object wrapped by the Dask DataFrame. That said, I still prefer the `engine` name to the `read_function` name. Do others have thoughts?

Reply:
Thanks for your help @TomAugspurger  - I mostly agree.  However, the differences between this argument and the one in `read_parquet` make `engine` feel a bit *off* to me.  I do like the term `engine` better, so perhaps I'm overthinking this?

Reply:
Agreed that `engine` is a bit off. Do we forsee supporting something like

```
dd.read_json(engine='pygdf')
```

then Dask would optionally import pygdf and attempt to use its read_json? If so, then I think I'm ok with the keyword `engine`.

Reply:
My motivation for this PR was to allow a gpu user (or `dask-cudf`) to use utilize `dask.dataframe` json functionality,  but with `cudf.read_json(...)` replacing `pd.read_json(...)`.  

I wanted to accomplish this in a minimally-intrusive way, such that dask is not responsible for handling the behavior of specific engines (besides `pd.read_json`).  I assumed that it is best to avoid the need to maintain a list of "implemented engines" unless this proves necessary.

Reply:
Makes sense.

I think given all that my preference for the keyword name is still `engine`. Despite it being a bit different from the parquet engine, it still has the same meaning.